src/githubFactory.js

```diff
@@ -1,111 +1,117 @@
-  });
-
-  // Transform a github function call into a queued function call
-  // to ensure that only one API call runs at a time
-  // https://developer.github.com/guides/best-practices-for-integrators/#dealing-with-abuse-rate-limits
-  var qd = function(call) {
-    if (typeof call !== "function")
-      throw new Error("call should be a function: " + call);
-    return function() {
-      var args = [].slice.apply(arguments);
-      return new Promise(function(resolve, reject) {
-        q.push(function(cb) {
-          var argsCb = args.pop();
-          if (typeof argsCb !== "function") {
-            args.push(argsCb);
-            argsCb = null;
-          }
-          call.apply(null, args).then(
-            function(result) {
-              cb();
-              if (argsCb) argsCb(null, result);
-              resolve(result);
-            },
-            function(error) {
-              cb();
-              if (argsCb) argsCb(error);
-              reject(error);
-            }
-          );
-        });
-        q.start();
-      });
-    };
-  };
-
-  var makeResponseTransformer = function(transform) {
-    return function(call) {
-      if (typeof call !== "function")
-        throw new Error("call should be a function: " + call);
-      return function() {
-        var args = [].slice.apply(arguments);
-        var argsCb = args.pop();
-        if (typeof argsCb !== "function") {
-          args.push(argsCb);
-          argsCb = null;
-        }
-        return call.apply(null, args).then(
-          function(result) {
-            result = transform(result);
-            if (argsCb) argsCb(null, result);
-            return result;
-          },
-          function(err) {
-            if (argsCb) argsCb(err);
-          }
-        );
-      };
-    };
-  };
-
-  // Unwrap .data in the response
-  var nd = makeResponseTransformer(function(response) {
-    return response.data;
-  });
-
-  // Add a warning about subtly invalid params
-  var hw = function(f) {
-    return function() {
-      var args = arguments;
-      if (args[0] && args[0].url) throw new Error("Avoid passing url option");
-      return f.apply(null, args);
-    };
-  };
-
-  var paginate = function(f) {
-    return function() {
-      return api.paginate(f, ...arguments);
-    };
-  };
-
-  return {
-    repos: {
-      compareCommits: hw(nd(qd(api.repos.compareCommits))),
-      delete: hw(nd(qd(api.repos.delete))),
-      get: hw(nd(qd(api.repos.get))),
-      list: hw(qd(paginate(api.repos.listForAuthenticatedUser))),
-      listBranches: hw(nd(qd(api.repos.listBranches)))
-    }
-  };
+module.exports = token => {
+	const api = rawGithubFactory(token);
+
+	const q = queue({
+		concurrency: 1,
+		timeout: 2000
+	});
+
+	// Transform a github function call into a queued function call
+	// to ensure that only one API call runs at a time
+	// https://developer.github.com/guides/best-practices-for-integrators/#dealing-with-abuse-rate-limits
+	const qd = call => {
```

Comment:
FYI I hope to replace this in a future PR; I think [p-queue](https://github.com/sindresorhus/p-queue) can replace both `qd` and the `queue` module while also helping a possible (internal) promisification of `remove-github-forks`. Callback hell can be avoided nowadays

Reply:
I fully support that.

Reply:
We’re on Node, so [util.promisify](https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original) can help transition a lot of these without rewriting everything in one go.